scipy/stats/morestats.py

```diff
@@ -3478,3 +3478,74 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
         R = np.minimum(1, hypot(sin_mean, cos_mean))
 
     return ((high - low)/2.0/pi) * sqrt(-2*log(R))
+
+
+def quartile_coeff_dispersion(samples, q=(0.25, 0.75), axis=None, interpolation='lower'):
+    """
+    Compute the quartile coefficient of dispersion for a given sample array.
+
+    Given two quartiles Q1, Q2 from array samples such that Q1 < Q2, the quartile
+    coefficient of dispersion is calculated by (Q2 - Q1)/(Q1 + Q2).
+    by default it is computed using 3rd and 1st (Q2, Q1 respectively) quartiles.
+
+    Parameters
+    ----------
+    samples : array_like
+        Input array.
+    q : array_like of float, optional
+        Sequence of 2 quartiles, one of [0.25, 0.5, 0.75].
+        The default is to compute with ￿Q2 = 0.75 (upper quartile), Q1 = 0.25 (lower
+        quartile).
+    axis : {int, tuple of int, None}, optional
+        Axis or axes along which the quantiles are computed. The
+        default is to compute the quartile(s) along a flattened
+        version of the array.
+    interpolation : {'linear', 'lower', 'higher', 'midpoint', 'nearest'}
+        This optional parameter specifies the interpolation method to
+        use when the desired quantile lies between two data points
+        ``i < j``:
+
+            * linear: ``i + (j - i) * fraction``, where ``fraction``
+              is the fractional part of the index surrounded by ``i``
+              and ``j``.
+            * lower: ``i``.
+            * higher: ``j``.
+            * nearest: ``i`` or ``j``, whichever is nearest.
+            * midpoint: ``(i + j) / 2``.
+
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Quartile_coefficient_of_dispersion
+
+    Returns
+    -------
+    dispersion : {float, array_like of float}
+        Dispersion for each axis.
```

Comment:
```suggestion
        Dispersion for each axis slice.
```
I'm not sure that "for each axis" is strictly accurate (e.g. there are many more elements in the array than there are axes). There is some precedent for using "axis slice" for what this is saying, though.